pkg/pillar/cmd/zfsmanager/zfsmanager.go

```diff
@@ -305,6 +309,17 @@ func deviceWatcher(ctxPtr *zfsContext) {
 		})
 	}
 
+	// Wait for directory to appear
```

Comment:
I think you could also use fsnotify to wait for the directory creation instead of the busy loop... something like this it works:

```go
func waitForDirCreation(path string) error {
  watcher, err := fsnotify.NewWatcher()
  if err != nil {
    return err
  }
  defer watcher.Close()

  wpath := filepath.Dir(path)
  err = watcher.Add(wpath)
  if err != nil {
    return err
  }
  for {
    select {
      case event, ok := <-watcher.Events:
      if !ok {
        return fmt.Errorf("Error watching directory ", wpath)
      }
      if event.Op == fsnotify.Create {
        if event.Name == path {
          return nil;
        }
      }
    }
  }
}
```

Reply:
Sure, we could do that with path=/dev, but that assumes udev doesn't delete and recreate /dev (I have no idea what it does and why).

Reply:
it doesn't delete /dev, we will always have basic file devices there, you don't need to worry about it.... your latest version looks good to me....

Reply:
Done.